consumer/src/pkce/pkce_code_challenge_method_sha256.e

```diff
@@ -0,0 +1,56 @@
+note
+	description: "Object representing code challenge sha256"
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	PKCE_CODE_CHALLENGE_METHOD_SHA256
+
+inherit
+
+	PKCE_CODE_CHALLENGE_METHOD
+
+	SHARED_BASE64
+
+feature -- Accesz
+
+
+	transform_code_challenge (a_code_verifier: STRING): STRING
+		local
+			l_str: STRING
+		do
+				--base64url
+				--https://tools.ietf.org/html/rfc7636#appendix-A
+			Result := base64_encoder.encoded_string (sha256_string (a_code_verifier))
+			Result.replace_substring_all ("+", "-")
```

Comment:
I guess using a loop over the string to replace + by - and so on would be more efficient
only 1 traversal instead of 3

Reply:
Updated code, using a loop to replace (`+` `,-`) and (`/`,`_`), keeping the call to prune_all_trailing to keep the code simple.